godot-macros/src/util/mod.rs

```diff
@@ -288,3 +288,65 @@ pub fn make_virtual_tool_check() -> TokenStream {
         }
     }
 }
+
+pub enum ViaType {
+    Struct,
+    EnumWithRepr(Ident),
+    Enum,
+}
+
+impl ToTokens for ViaType {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        match self {
+            ViaType::Struct => quote! { ::godot::builtin::Variant }.to_tokens(tokens),
+            ViaType::EnumWithRepr(id) => id.to_tokens(tokens),
+            ViaType::Enum => quote! { ::godot::builtin::Variant }.to_tokens(tokens),
+        }
+    }
+}
+
+pub fn via_type(declaration: &venial::Declaration) -> ParseResult<ViaType> {
+    use venial::Declaration;
+
+    match declaration {
+        Declaration::Enum(enum_) => enum_repr(enum_),
+        Declaration::Struct(_) => Ok(ViaType::Struct),
+        other => bail!(
+            other,
+            "cannot get via type for {:?}, only structs and enums are supported currently",
+            other.name()
+        ),
+    }
+}
+
+pub fn enum_repr(enum_: &venial::Enum) -> ParseResult<ViaType> {
+    let Some(repr) = enum_
+        .attributes
+        .iter()
+        .find(|attr| attr.get_single_path_segment() == Some(&ident("repr")))
+    else {
+        return Ok(ViaType::Enum);
+    };
+
+    let venial::AttributeValue::Group(_, repr_value) = &repr.value else {
+        // `repr` is always going to look like `#[repr()]`
+        unreachable!()
```

Comment:
any form of `#[repr]` that doesn't look like `#[repr()]` will give a rust compile time error:
```
#[repr = i64] // attribute value must be a literal
#[repr = "i64"] // malformed `repr` attribute input
#[repr] // malformed `repr` attribute input
#[repr(i64, i32)] // conflicting representation hints, this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
```

however `#[repr()]` is just a warning apparently:
```
#[repr()] // unused attribute, attribute `repr` with an empty list has no effect 
```
which surprised me a bit tbh, i'll fix that then ig,

Reply:
This is the only thing left, otherwise I think it's ready 🙂